src/toil_scripts/gatk_germline/germline.py

```diff
@@ -389,50 +335,187 @@ def gatk_haplotype_caller(job, bam_id, bai_id, config, emit_threshold=10.0, call
+    :param str uuid: Unique identifier for the sample
+    :param str url: URL to BAM or FASTQ file
+    :param Namespace config: Configuration options for pipeline
+    :param str|None url2: URL to paired FASTQ file
+    :param str|None rg_line: RG line for BWA alignment. Default is None.
+    :return: None
+    """
+    get_shared_files = job.wrapJobFn(download_shared_files, config).encapsulate()
+    run_pipeline = Job.wrapJobFn(run_gatk_germline_pipeline,
+                                 [(uuid, url, url2, rg_line)],
+                                 get_shared_files.rv())
+    get_shared_files.addChild(run_pipeline)
+
+
+def run_gatk_germline_pipeline(job, samples, config):
+    """
+    Configures and runs the GATK germline pipeline and filters the output. VQSR is done if the
+    run_vqsr config parameter is set to True or there are at least 30 samples. If joint genotyping
+    is done, the VCF is split per sample after filtering. Otherwise, each sample VCF is hard
+    filtered using recommended GATK parameters.
+
+    :param JobFunctionWrappingJob job: passed automatically by Toil
+    :param list samples: List of tuples (uuid, url, url2, rg_line) for joint genotyping.
+    :param Namespace config: Input parameters and reference FileStoreIDs
+    :return: Dictionary of filtered VCF FileStoreIDs
+    :rtype: dict
+    """
+    if len(samples) == 0:
+        raise ValueError('No samples were provided!')
+
+    # Generate per sample gvcfs {uuid: gvcf_id}
+    gvcfs = {}
+    for uuid, url, url2, rg_line in samples:
+        gvcfs[uuid] = job.addChildJobFn(gatk_germline_pipeline,
+                                        uuid,
+                                        url,
+                                        config,
+                                        url2=url2,
+                                        rg_line=rg_line).rv()
+
+    # Stop after preprocessing
+    if config.preprocess_only:
+        return
+
+    # VQSR requires many variants in order to train a decent model. GATK recommends a minimum of
+    # 30 exomes or one large WGS sample:
+    # https://software.broadinstitute.org/gatk/documentation/article?id=3225
+    filtered_vcfs = {}
+    if config.run_vqsr or len(gvcfs) > 30:
+        if config.joint:
+            joint_vcf = job.addFollowOnJobFn(vqsr_pipeline, gvcfs, config)
+            filtered_vcfs = joint_vcf.addChildJobFn(split_vcf_by_name, gvcfs.keys(),
+                                                    joint_vcf.rv(), config).rv()
+        else:
+            for uuid, gvcf in gvcfs.iteritems():
+                filtered_vcfs[uuid] = job.addFollowOnJobFn(vqsr_pipeline,
+                                                           dict(uuid=gvcf),
+                                                           config).rv()
+    else:
+        for uuid, gvcf in gvcfs.iteritems():
+            filtered_vcfs[uuid] = job.addFollowOnJobFn(hard_filter_pipeline,
```

Comment:
We don't need to address this here (read: maybe open a ticket for this?) but it would probably make sense to support joint calling _followed_ by hard filtering.